### PR TITLE
specification: desktop extension font hook

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: specifications/desktop-extensions-font-hook.org [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+The =fonts.conf= file shall be modified to achieve fallback
+mechanisms.
+
+When the =configure= hook runs it must run =fc-cache= to update the fonts
+cache with from fonts that are generally available on the system using
+a privately generated =fonts.conf=. This cache must reside in a location
+that is generally available to the snap when run by a desktop user, to
+achieve such result, the cache shall reside in =$SNAP_DATA=.
+
+When the desktop appliaction launches, the launcher generated by the
```
Comment (cjp256) [inline specifications/desktop-extensions-font-hook.org:29]:
```suggestion
When the desktop application launches, the launcher generated by the
```

File: specifications/desktop-extensions-font-hook.org [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+extension shall contain a reference to this cache and only process
+fonts that are installed on the user's private font instance, =~/.fonts=.
```
Comment (cjp256) [inline specifications/desktop-extensions-font-hook.org:31]: Is .fonts accessible as a result of the desktop interface?

File: specifications/desktop-extensions-font-hook.org [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+The YAML generated by the desktop extensions shall be /extended/ to add
+a =hook= entry that references this new =command-chain=, such that the
+following is generated:
+
+#+BEGIN_SRC yaml
+hooks:
+  configure:
+    command-chain:
+    - snap/command-chain/snapcraft-runner
+    - snap/command-chain/hooks-configure
```
Comment (cjp256) [inline specifications/desktop-extensions-font-hook.org:59]:
```suggestion
    - snap/command-chain/hooks-configure-desktop-fonts
```

Comment (sergiusens) [reply specifications/desktop-extensions-font-hook.org:59]:
```suggestion
    - snap/command-chain/hooks-configure-desktop
```

File: specifications/desktop-extensions-font-hook.org (hunk 2 above)
Comment (sergiusens) [reply specifications/desktop-extensions-font-hook.org:31]: Desktop interface is required to reach the system fonts

**Review by cjp256 — APPROVED** [review]
👍

**Pull request merged**